src/docs/user/references/Variables.adoc

```diff
@@ -0,0 +1,408 @@
+
+The SpEL expression must return a *boolean value*, `true` if the value is correct, `false` otherwise, returning another type will raise an error.
+
+* *Example of SpEL simple validations:* +
++
+PA:SPEL(#value == 'abc') : will accept the value if it's the 'abc' string +
+PA:SPEL(T(Integer).parseInt(#value) > 10) : will accept the value if it's an integer greater than 10. +
++
+Note that #value always contains a string and must be converted to other types if needed. +
++
+* *Example of SpEL multiple validations:* +
++
+PA:SPEL(variables['var1'] + variables['var2'] == 'abcdef') : will be accepted if the string concatenation of variable var1 and var2 is 'abcdef'. +
+PA:SPEL(T(Integer).parseInt(variables['var1']) + T(Integer).parseInt(variables['var2']) < 100) : will be accepted if the sum of variables var1 and var2 are smaller than 100. +
++
+* *Example of SpEL variable inference:* +
++
+PA:SPEL( variables['var2'] == '' ? (variables['var2'] = variables['var1']) instanceof T(String) : true ) : if the variable var2 is empty, it will use the value of variable var1 instead. +
++
+Note that the SpEL expression must always return a boolean value, this is why in this last expression we use the instanceof keyword to avoid returning a string value.
+
+
+
+===== Generic Information
+
+In addition to variables, another key/value structure can be accessed inside a script: *_Generic Information_*.
+
+Generic information semantics differ from <<_job_variables>> semantics in the following way:
+
+ * Generic information can be accessed inside a script, but cannot be modified.
+ * Generic information can be defined at *job level* or at *task level*. If the same generic information is defined at job level and at task level, the latter value takes precedence inside the task scope.
+ * Generic information cannot be used directly inside the workflow with the syntax ${} (See  <<_job_variables>>).
+ * Generic information are used in general internally by the scheduler, for example to provide information to the scheduling loop on how to handle the task. An example of such generic information is the _START_AT_ info used inside <<_cron_tasks>>.
+ * Generic information can use in their definition job/task variables patterns, pattern replacements can be done at execution time when using task generic information.
+
+Example of generic information definition:
+
+For example:
+[source, xml]
+----
+<task ... >
+    <genericInformation>
+        <info name="ginfo" value="${test}"/>
+    </genericInformation>
+    ...
+</task>
+----
+
+[[_dynamic_variables]]
+
+==== Dynamic Variables
+
+As opposed to the <<_workflow_variables>>, *dynamic variables* are all variables created outside the XML job definition.
+We have mainly two types of dynamic variables :
+
+- <<_system_variables>>, declared by the ProActive scheduller.
+- <<_inherited_variables>>, created after adding values to the *variables* map.
```

Comment:
Created or manipulated directly when executing scripts, through the use of the "variables" map (make a link to the variables quick reference)

Reply:
ok